### PR TITLE
Fix Okta authentication Deadlock issue 

### PR DESCRIPTION
**Pull request by ChTimTsubasa**
This is a fix to address #138 
Have separate code routine to do sync authentication;
Using threadpool for execution.

### CODE REVIEW

File: Snowflake.Data.Tests/SFConnectionIT.cs [hunk 1]
```diff
@@ -92,9 +92,9 @@ public void TestLoginTimeout()
                     conn.Open();
                     Assert.Fail();
                 }
-                catch(SnowflakeDbException e)
+                catch (AggregateException e)
                 {
-                    Assert.AreEqual(SFError.REQUEST_TIMEOUT.GetAttribute<SFErrorAttr>().errorCode,e.ErrorCode);
+                    Assert.AreEqual(270007, ((SnowflakeDbException)e.InnerException).ErrorCode);
```
Comment (smtakeda) [inline Snowflake.Data.Tests/SFConnectionIT.cs]: Use const variable. Don't we have a list of error code for .NET driver?

**Comment by smtakeda** [comment]
Add test cases for OKTA auth.

**Comment by howryu** [comment]
Can you elaborate a little bit more what cause the issue and what is the fix? It seems that you just add a synchronous `Authenticate` method? Why async does not work?

**Comment by ChTimTsubasa** [comment]
@howryu Basically I was calling .Result, which will definitely result in deadlock. Based on some research (especially https://medium.com/rubrikkgroup/understanding-async-avoiding-deadlocks-e41f8f2c6f5d), As a library code, our best practice would be:
1. implementing both async and sync method, and separate code routine for best performance;
2. Implement our own syncContext and TaskScheduler to avoid deadlock in any circumstances.

This PR does not include 2 given that it is too much work and this PR should go fast to unblock users. But Task.Run() is used to mitigate deadlock issue. Another JIRA would be open to really finish 2.

**Comment by ChTimTsubasa** [comment]
@smtakeda There is already a manual test called TestOktaConnection which would test against OKTA authentication on local reg. Maybe we can consider having automated native OKTA test on Prod server.

File: Snowflake.Data.Tests/SFConnectionIT.cs (hunk 1 above)
Comment (ChTimTsubasa) [reply Snowflake.Data.Tests/SFConnectionIT.cs]: Addressed

File: Snowflake.Data.Tests/Mock/MockCloseSessionGone.cs [hunk 2]
```diff
@@ -29,6 +29,11 @@ public Task<HttpResponseMessage> GetAsync(IRestRequest request, CancellationToke
...
         }
 
+        public HttpResponseMessage Get(IRestRequest request)
+        {
+            return null;
+        }
+
         public T Post<T>(IRestRequest postRequest)
         {
             return Task.Run(async () => await PostAsync<T>(postRequest, CancellationToken.None)).Result;
```
Comment (howryu) [inline Snowflake.Data.Tests/Mock/MockCloseSessionGone.cs]: So in theory this could cause deadlock as well right?

Comment (ChTimTsubasa) [reply Snowflake.Data.Tests/Mock/MockCloseSessionGone.cs]: Yes

**Comment by howryu** [comment]
LGTM. That's really good article to read about async coding in .NET.

**Pull request merged**